OpenRA.Game/FileSystem/GlobalFileSystem.cs

```diff
@@ -185,10 +185,23 @@ public static Stream Open(string filename)
 			return s;
 		}
 
-		public static bool TryOpen(string filename, out Stream s)
+		public static bool TryOpen(string name, out Stream s)
 		{
-			// Check the cache for a quick lookup
-			if (filename.IndexOfAny(new char[] { '/', '\\' }) == -1)
+			var filename = name;
+			var foldername = string.Empty;
+
+			// Used for faction specific packages; rule out false positive on Windows C:\ drive notation
+			var explicitFolder = name.Contains(':') && !Directory.Exists(Path.GetDirectoryName(name));
+			if (explicitFolder)
+			{
+				var divide = name.Split(':');
+				foldername = divide.First();
+				filename = divide.Last();
+			}
+
+			// Check the cache for a quick lookup if the folder name is unknown
+			// TODO: explicit folders will always take the slow path
```

Comment:
I replaced the `string.IsNullOrEmpty(folder)` with `if (explicitFolder)` which should be faster and essentially say that in code.